src/libunicode/u_str.rs

```diff
@@ -77,7 +77,7 @@ impl UnicodeStr for str {
     fn is_alphanumeric(&self) -> bool { self.chars().all(|c| c.is_alphanumeric()) }
 
     #[inline]
-    fn width(&self, is_cjk: bool) -> usize {
+    fn width(&self, is_cjk: bool) -> u32 {
```

Comment:
Hm, this seems like somewhat of an edge case: the width of a string will typically be fairly similar (at least, proportional) to its size in memory.